### ISSUE TITLE
[bug]: TypeError: Cannot read properties of undefined (reading 'filter') after upgrading to version 1.4.2

### ISSUE THREAD

**Issue by riadhmouamnia** [JaleelB/emblor#76]
### Describe the bug

After upgrading from version 1.4.0 to 1.4.2 of the emblor library, my application crashes with the following error:

```
TypeError: Cannot read properties of undefined (reading 'filter')
```

### How to reproduce

1. Install version 1.4.2 of the emblor library.
2. Set up a React component with the following code:

```jsx
import { Tag, TagInput } from "emblor";
import { useState } from "react";

function App() {
  const [tags, setTags] = useState<Tag[]>([]);
  const [activeTagIndex, setActiveTagIndex] = useState<number | null>(null);
  return (
    <main className="flex flex-col items-center justify-center h-screen">
      <p>Tag input:</p>
      <TagInput
        tags={tags}
        setTags={(newTags) => setTags(newTags)}
        placeholder="Add a tag"
        styleClasses={{
          input: "w-full sm:max-w-[350px]",
        }}
        activeTagIndex={activeTagIndex}
        setActiveTagIndex={setActiveTagIndex}
      />
    </main>
  );
}

export default App;
```

3. Run the application.

**Expected Behavior:**
The application should run without errors, and TagInput component should work as expected.

**Actual Behavior:**
The application crashes with the following error:

```
TypeError: Cannot read properties of undefined (reading 'filter')
```

### Link to reproduction

https://github.com/riadhmouamnia/emblor-bug

### Additional information

Upon inspecting the console, the error seems to be related to the filtering of `tag.text`. The relevant code snippet causing the error is:

```
reactExports.useMemo(() => _ ? f.filter((d) => _(d.text)) : f.filter((d) => d.text.toLowerCase().includes(O.toLowerCase())), [O, f, _])
```

**Screenshots:**
![image](https://github.com/user-attachments/assets/0ad8e6ed-39db-4e1d-a491-17ab2f753308)

It appears that the `filter` function is being called on an undefined value, potentially due to an issue with the `tag.text` property filtering.

**Environment:**
- React version: 18.3.1
- React DOM version: 18.3.1
- emblor version: 1.4.2
- tailwindcss: 3.3.2
- vite: 4.3.9

**Comment by JaleelB** [comment]
Hey @riadhmouamnia. Thanks for bringing this up. I am aware of this issue that was introduced in v1.4.2, and I am currently working on the fix!

**Comment by riadhmouamnia** [comment]
> Hey @riadhmouamnia. Thanks for bringing this up. I am aware of this issue that was introduced in v1.4.2, and I am currently working on the fix!

Great, good luck!

**Comment by JaleelB** [comment]
The issue discussed in this thread have been addressed in the [v1.4.4 release](https://github.com/JaleelB/emblor/releases/tag/emblor%401.4.4).

**Comment by riadhmouamnia** [comment]
> Hey @riadhmouamnia. The issue discussed in this thread have been addressed in the [v1.4.4 release](https://github.com/JaleelB/emblor/releases/tag/emblor%401.4.4).

Thank you, now the component works as expected and runs without errors.
